Set the C++ standard to C++0x everywhere

rwy0717 opened the pull request:
Signed-off-by: Robert Young <rwy0717@gmail.com>

charliegracie commented:
@genie-omr build all

charliegracie commented on cmake/modules/platform/toolcfg/gnu.cmake:
```diff
@@ -68,7 +70,7 @@ list(APPEND TR_COMPILE_OPTIONS
 # C++ file compilation
 list(APPEND TR_CXX_COMPILE_OPTIONS
-	-std=c++0x
+	# nothing
```
Can you remove the TR_CXX_COMPILE_OPTIONS?

rwy0717 replied:
done

charliegracie commented:
@genie-omr build all

rwy0717 commented:
Somehow I managed to add a backtick to the end of aix.cmake, fixed now. The linux builds failed to clone, seems there's an infrastructure issue? While I was at it, I rebased as well.

This is ready for a build.
